src/config/Config.py

```diff
@@ -29,53 +32,76 @@ def ingest_env(self):
         if "PIPELINE_DB_NAME" in os.environ:
             self._data["pipeline_db_name"] = os.environ["PIPELINE_DB_NAME"]
 
+    def _is_date(self, val):
+        if not isinstance(val, str):
+            return False
+
+        if len(val) == 10:
+            for i in range(len(val)):
+                if i == 4 or i == 7:
+                    if val[i] != '-':
+                        return False      
+                else:
+                    if not val[i].isnumeric():
+                        return False
+        else:
+            return False                
+        return True
+
+    def _is_na(self, val):
+        if val == 'NA' or val == 'na' or val == '':
+            return True
+        else:
+            return False
+
     def set_value(self, name, val):
-        self._data[name] = val
+        if not name == 'columns':
+            self._data[name] = val
 
-    def get_value(self, val):
-        if val in self._data:
-            return self._data[val]
+    def get_value(self, name):
+        if not name == 'columns':
+            if name in self._data:
+                return self._data[name]
 
-    def check_bounds(self, column_name, val):
-        if "date" in column_name:
-            print(val)
-            val = parse(val)
+    def set_bounds(self, column_name, min, max):
+        self._data['columns'][column_name] = {'min' : min, 'max' : max}
+
+    def get_bounds(self, column_name):
+        if column_name in self._data['columns']:
+            return self._data['columns'][column_name]
 
+    def check_bounds(self, column_name, val):
         if column_name in self._data["columns"]:
             col = self._data["columns"][column_name]
 
             col_max = None
             col_min = None
 
-            if "date" in column_name:
-                if col['max'] != 'NA':
-                    col_max = parse(col["max"])
-                else:
-                    col_max = col['max']
+            if self._is_date(val):
+                val = parse(val)
+                if not self._is_na(col['max']):
```

Comment:
So what happens when col['max'] doesn't exist? I suspect it might give you a keyerror.